### ISSUE TITLE
How to suppress printing execution time using $this->taskExec() ?

### ISSUE THREAD

**Issue by disharmonized** [consolidation/Robo#383]
How to suppress printing execution time using `$this->taskExec()`? I have my own implemented execution time counter and i want to use it, not robo's version. Are there any opportunities to achieve that?

I am using 1.0.0beta1 robo's version.

**Comment by greg-1-anderson** [comment]
Robo prints success/error results, including execution time, from the constructor of the Result object.  This gives clients no opportunity to suppress this output.

Ideally, this would be refactored in a future version of Robo. In the 1.x timeframe, the best we could do would be to provide some sort of static variable to disable the result output. I'm not sure if that would be worth the overhead.

Could you provide more details on your use case?  e.g. What would your desired output (including your custom time counter output) look like?

**Comment by disharmonized** [comment]
Ok, here's my case:

```php
class RoboFile extends \Robo\Tasks
{
    public function exportFromMySql()
    {
        $taskStartTime = time();

        $sql = 'select * from bigtable;';
        $output = '~/output/out.csv';

        $this->taskExec('mysql')
            ->arg('testdb')
            ->arg('--host=localhost')
            ->arg('--port=1234')
            ->arg('--user=test')
            ->arg('--password=test')
            ->arg('--quick')            
            ->arg(' -c "' . $sql . '" > ' . $output)
            ->printed(true)
            ->run()
            ->stopOnFail();

        //it prints execution time like
        //Done for 05:10:45
        $this->say('Done for ' . Util::getFormattedExecutionTime($taskStartTime, time()));

        //and after this robo prints it's own execution time output, which is already unnecessary for me
        
    }

    public function importToMongoDb(){
        //similar task
    }
}
```

It will be very handy to have opprotunity to change behaviour of Result output, i.e disable it, format it, whatever.

**Comment by disharmonized** [comment]
>  In the 1.x timeframe, the best we could do would be to provide some sort of static variable to disable the result output
I think that will be enough for me.

**Comment by cyberhorse** [comment]
Static variable is fine. It should disable both the starting and ending output.

**Comment by greg-1-anderson** [comment]
I refactored the Config class to make it easier to reference from other classes (Config no longer contains a reference to the Container).  TaskIO now checks Config::isSuppressed() before doing output.  Things emitted by `say` are not affected.

Some things I realized in working on that PR:

- Tasks should not care about output; they should just use TaskIO.
- Robo commands should not change the default output formatting; they should rely on the application to do that.
- Applications that are built on top of Robo can initialize up their own DI container if they would like to change the way things work (e.g. output). Modifying the container  has to happen at the application layer, because a lot of objects will already have been created by the time the command runs, by which time it will be too late to modify most of the shared (singleton) objects in the container.

As far as this feature request is concerned, I think it would be better to submit a PR to robo to change the formatting of the execution time universally rather than try to turn off the default output and supply your own preferred time formatter.
